tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Redis/RedisTags.cs

```diff
@@ -25,5 +25,12 @@ internal partial class RedisTags : InstrumentationTags
 
         [Tag(Trace.Tags.OutPort)]
         public string Port { get; set; }
+
+        // Always use metrics for "number like" tags. Even though it's not really a "metric"
+        // that should be summed/averaged, it's important to record it as such so that we
+        // don't use "string-like" searching. e.g. searching for db.redis.database_index=1
+        // should only bring back 1, not 10 or 11 as it would otherwise.
```

Comment:
I think the issue is more with sorting, since searching for `1` is an exact match, unlike `1*`, no?

sorting numbers as strings:
```
1
10
2
20
```